relay-metrics/src/aggregation.rs

```diff
@@ -1782,22 +1785,35 @@ impl Aggregator {
             let partitioned_buckets = self.partition_buckets(project_buckets, num_partitions);
             for (partition_key, buckets) in partitioned_buckets {
                 self.process_batches(buckets, |batch| {
-                    let batch_size = batch.len();
-                    match self.receiver.do_send(FlushBuckets {
+                    if let Err(err) = self.receiver.do_send(FlushBuckets {
                         project_key,
                         partition_key,
                         buckets: batch,
                     }) {
-                        Ok(_) => {}
-                        Err(err) => {
-                            relay_log::error!("Failed to flush the buckets: {err}. Dropping current bucket batch of size: {batch_size}");
-                        }
+                        let FlushBuckets {
+                            project_key,
+                            buckets,
+                            ..
+                        } = err.into_inner();
+
+                        // remove the number of buckets which were not flushed due to the `SendError`
+                        total_bucket_count -= buckets.len() as u64;
+                        buckets_failed.push((project_key, buckets));
                     }
                 });
             }
         }
-
         relay_statsd::metric!(histogram(MetricHistograms::BucketsFlushed) = total_bucket_count);
+
+        relay_log::trace!(
```

Comment:
I think we can keep log level `error!` here (if any buckets failed). A `SendError` is not something we expect AFAIK.